src/esn.calendar.libs/app/components/event-preview-popover/event-preview-popover.controller.js

```diff
@@ -0,0 +1,194 @@
+angular.module('esn.calendar.libs')
+  .controller('CalEventPreviewPopoverController', CalEventPreviewPopoverController);
+
+function CalEventPreviewPopoverController(
+  $log,
+  $scope,
+  $modal,
+  session,
+  calEventPreviewPopoverService,
+  calendarService,
+  calAttendeeService,
+  calOpenEventForm,
+  calEventUtils,
+  calEventService,
+  calEventDuplicateService,
+  calPartstatUpdateNotificationService,
+  calUIAuthorizationService,
+  urlUtils,
+  notificationFactory
+) {
+  const self = this;
+
+  self.openEventForm = openEventForm;
+  self.deleteEvent = deleteEvent;
+  self.duplicateEvent = duplicateEvent;
+  self.changeParticipation = changeParticipation;
+  self.closeEventPreviewPopover = closeEventPreviewPopover;
+
+  function openEventForm() {
+    closeEventPreviewPopover();
+
+    calOpenEventForm(self.event.calendarHomeId, self.event.clone());
+  }
+
+  function deleteEvent() {
+    closeEventPreviewPopover();
+
+    if (!self.event.isInstance()) {
+      return calEventService.removeEvent(self.event.path, self.event, self.event.etag, false);
+    }
+
+    $modal({
+      template: require('../event/form/modals/delete-instance-or-series-modal.pug'),
+      controller: /* @ngInject */ function($scope) {
+        $scope.editChoice = 'this';
+
+        $scope.submit = function() {
+          $scope.$hide();
+
+          $scope.editChoice === 'this' ? deleteInstance() : deleteMaster();
+        };
+
+        function deleteMaster() {
+          calEventService.removeEvent(self.event.path, self.event, self.event.etag, true);
+        }
+
+        function deleteInstance() {
+          calEventService.removeEvent(self.event.path, self.event, self.event.etag, false);
+        }
+      },
+      placement: 'center'
+    });
+  }
+
+  function duplicateEvent() {
+    calEventDuplicateService.duplicateEvent(self.event).then(duplicatedEvent => {
+      calOpenEventForm(session.user._id, duplicatedEvent);
+    });
+  }
+
+  function changeParticipation(status) {
```

Comment:
so dont using a generic changeParticipation with event-form

Reply:
Good question. If you look closer, this code actually looks a little bit different from the code used in the event form controller. But I'll try to make the code reusable some time later because it'd be a little bit complicated to do it right now.